mods/default/functions.lua

```diff
@@ -293,13 +293,41 @@ minetest.register_abm({
 -- Dig upwards
 --
 
-function default.dig_up(pos, node, digger)
+local in_dig_up = false
+
+---Find all nodes above this one that is the same, then dig them all
+---@param pos vector The position of the base node
+---@param node { name: string, param1: integer, param2: integer } Node table of the base node
+---@param digger ObjectRef The object (e.g. player) digging the node
+---@param max_height The maximum height to search for, excluding the base node
+function default.dig_up(pos, node, digger, max_height)
+	if in_dig_up then return end -- Do not recurse
 	if digger == nil then return end
-	local np = {x = pos.x, y = pos.y + 1, z = pos.z}
-	local nn = minetest.get_node(np)
-	if nn.name == node.name then
-		minetest.node_dig(np, nn, digger)
+	max_height = max_height or 100
+
+	in_dig_up = true
+	for y = pos.y + 1, pos.y + max_height do
+		local up_pos  = vector.new(pos.x, y, pos.z)
+		local up_node = minetest.get_node(up_pos)
+		if up_node.name ~= node.name then
+			break
+		end
+		local noerr, success = xpcall(function()
+			minetest.dig_node(up_pos, digger)
+		end, function(...)
+			in_dig_up = false
+			minetest.log("error", "Error raised during `default.dig_up` call:")
+			for line in debug.traceback(...):gmatch("([^\n]*)\n?") do
+				minetest.log("error", line)
+			end
```

Comment:
As an alternative to manual parsing and outputting - might `minetest.error_handler(...)` produce better output?

Reply:
Weirdly, using `minetest.error_handler` does not print anything about the error... IDK why.

Reply:
`minetest.error_handler` should give you a formatted error string you can log, it doesn't do the printing itself.

Reply:
Done in b49153c